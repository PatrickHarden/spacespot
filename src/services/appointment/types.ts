export interface AppointmentState {
  data?: Array<Appointment>
  error?: string
}

export interface Appointment {
  id: number
  enquireId: number
  description: string
  peer1Id: string
  peer2Id: string
  proposedTime: string
  createdAt: string
  lastUpdatedAt: string
  lastProposedBy: string
  appointmentStatus: string
}

export interface NewAppointment {
  description: string
  proposedTime: string
}
export interface EditAppointment {
  id: string
  description: string
  proposedTime: string
}
export interface AppointmentAction {
  type: string
  payload?: Array<Appointment> | NewAppointment | EditAppointment | string
}
