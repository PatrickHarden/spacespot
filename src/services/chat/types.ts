export interface ChatAction {
  type: string
  payload?: ChatResponse | ChatMessage | string
}
export interface MessageAction {
  type: string
  payload?: string
}
export interface ChatState {
  data?: ChatResponse
  error?: string
}

export interface ChatResponse {
  chatId: number
  enquiryId: number
  peer1Id: string
  peer2Id: string
  messages: Array<ChatMessage>
}

export interface ChatMessage {
  userId: string
  message: string
  sentAt: string
}
