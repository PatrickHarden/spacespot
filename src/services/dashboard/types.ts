import { Space } from 'services/space/types'

export interface DashboardState {
  buildings: Array<Space>
  spaces: { [key: string]: Array<Space> }
  error?: string
  isLoading: boolean
  initOK: boolean
  pendingDelete: Array<string>
}
