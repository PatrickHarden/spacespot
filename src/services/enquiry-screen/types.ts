export interface EnquiryScreenAction {
  type: string
  payload: EnquiryScreenStateData | string
}
export interface EnquiryScreenState {
  data?: EnquiryScreenStateData
  error?: string
}
export interface EnquiryScreenStateData {
  id: string
}
