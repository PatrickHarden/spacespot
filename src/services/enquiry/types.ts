import { FileItem } from 'services/global/types'
import { NegotiationTerm } from 'services/negotiation/types'

export interface DocumentDTO {
  fileId: number
  fileName: string
  enquireId: number
  uploadAt: Date
  uploadedBy: string
  documentType?: string
}

export const DOC_TYPE_TEMPLATE = 'CompanyTemplate'

export type DocData = any

export interface EnquiryAction {
  type: string
  payload:
    | string
    | number
    | DocumentDTO
    | Array<DocumentDTO>
    | EnquiryResponse
    | FileItem
    | DocData
    | EnquiryRequest
}

export interface EnquiryState {
  data?: EnquiryResponse
  docs?: Array<DocumentDTO>
  error?: string
}

export enum EnquiryType {
  HotDesk = 'HotDesk',
  FixedDesk = 'FixedDesk',
  ServicedOffice = 'ServicedOffice',
  notChecked = '',
}

export interface EnquiryResponse {
  chatId: number
  enquireId: number
  spaceId: string
  /* Tenant */
  peer1Id: string
  /* Landlord */
  peer2Id: string
  negotiationId?: number
  enquireStatus?: string
  flexType?: EnquiryType
  flex: boolean
}

export interface EnquiryAPIResultActions {
  successSignInAfter?: boolean
  errorShowMessage?: string
}

export interface EnquiryRequest {
  spaceId: string
  message: string
  negotiation: {
    flexible: boolean
    terms: Array<NegotiationTerm>
  }
  userId?: string
  loggedIn?: boolean
  apiResultActions?: EnquiryAPIResultActions
}

export interface EnquiryRequestFlex {
  spaceId: string
  message: string
  flex: boolean
  flexType: EnquiryType
  flexIdentifier?: string
  negotiation: {
    terms: Array<NegotiationTerm>
  }
  userId?: string
  loggedIn?: boolean
  apiResultActions?: EnquiryAPIResultActions
}
