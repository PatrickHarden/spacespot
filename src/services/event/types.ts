export interface EnquiryEvent {
  eventId: number
  eventMessage: any
  eventType: string
  eventDate: Date
}

export interface EnquiryEventState {
  [enquiryId: string]: {
    events: Array<EnquiryEvent>
    isLoading?: boolean
    error?: string
  }
}

export interface GetEnquiryEventsAction {
  type: string
  payload: {
    enquiryId: string
  }
}

export interface GetEnquiryEventsSuccessAction {
  type: string
  payload: {
    enquiryId: string
    events: Array<EnquiryEvent>
  }
}

export interface GetEnquiryEventsErrorAction {
  type: string
  payload: {
    enquiryId: string
    error: string
  }
}

export type EnquiryEventAction =
  | GetEnquiryEventsAction
  | GetEnquiryEventsSuccessAction
  | GetEnquiryEventsErrorAction
