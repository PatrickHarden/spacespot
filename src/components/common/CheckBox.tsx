import React from 'react'
import styled from 'styled-components'

const CheckBox = styled(
  (props: {
    children: React.ReactNode
    className?: string
    checked?: boolean
    onChange?: (event: React.ChangeEvent<HTMLInputElement>) => void
  }) => {
    return (
      <label className={props.className}>
        {props.children}
        <input
          type="checkbox"
          checked={props.checked}
          onChange={props.onChange}
        />
        <span></span>
      </label>
    )
  },
)`
  display: block;
  position: relative;
  cursor: pointer;
  padding-left: 25px;
  margin: 10px 0;
  input {
    position: absolute;
    opacity 0;
    cursor: pointer;
    width: 0;
    height: 0
  }
  span {
    position: absolute;
    top: 0;
    left: 0;
    height: 15px;
    width: 15px;
    border: 2px solid #B3DBE9;
    margin-top: 1px;
  }
  input:checked ~ span {
    background-color: transparent;
  }

  span:after {
    content: " ";
    position: absolute;
    display: none;
  }

  input:checked ~ span:after {
    display: block;
  }

  span:after {
    left: 5px;
    top: 2px;
    width: 5px;
    height: 8px;
    border: solid black;
    border-width: 0 1px 2px 0;
    transform: rotate(45deg);
  }
`

export default CheckBox
