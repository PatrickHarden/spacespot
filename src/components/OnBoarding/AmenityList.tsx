import React, { useEffect } from 'react'
import { useDispatch, useSelector } from 'react-redux'
import styled from 'styled-components'
import { cloneDeep } from 'lodash'

import { AmenityOptions } from 'services/onboarding/types'
import actions from 'services/onboarding/actions'
import selectors from 'services/onboarding/selectors'
import CheckBox from 'components/common/CheckBox'
import { getUserIntl } from 'intl'

const AmenityList = styled(
  (props: { className?: string; amenities: AmenityOptions }) => {
    const { className, amenities } = props
    const options = useSelector(selectors.defaultAmenities)
    const dispatch = useDispatch()

    const list: AmenityOptions = Object.keys(amenities).reduce(
      (obj: AmenityOptions, key) => {
        obj[key] = amenities[key]
        return obj
      },
      options ? cloneDeep(options) : {},
    )

    useEffect(() => {
      if (!options) {
        dispatch(actions.initDefaultAmenities())
      }
    }, [dispatch, options])

    const onChange = (key: string) => (
      event: React.ChangeEvent<HTMLInputElement>,
    ) => {
      const newAmenities = { ...list }
      newAmenities[key].checked = event.target.checked
      dispatch(actions.setAmenities(newAmenities))
    }

    const intl = getUserIntl()

    return (
      <div className={className}>
        {Object.keys(list).map(i => {
          return (
            <CheckBox onChange={onChange(i)} key={i} checked={list[i].checked}>
              {intl.formatMessage({ id: i.toUpperCase() })}
            </CheckBox>
          )
        })}
      </div>
    )
  },
)`
  min-height: 100px;
  margin-left: 10px;
  label {
    color: #828286;
  }
`

export default AmenityList
