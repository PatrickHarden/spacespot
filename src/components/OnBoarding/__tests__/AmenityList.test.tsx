import React from 'react'
import { mount } from 'enzyme'
import { mountToJson } from 'enzyme-to-json'

import AmenityList from '../AmenityList'
import CheckBox from 'components/common/CheckBox'
import { SpacespotState } from 'services/global/types'

const mockDispatch = jest.fn()
const mockState = {}
jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  useDispatch: () => mockDispatch,
  useSelector: (selector: (state: SpacespotState) => any) =>
    selector(mockState),
}))

it('renders without crashing ', async () => {
  const amenities = {
    op1: { desc: 'op1', checked: true },
  }
  const wrapper = mount(<AmenityList amenities={amenities} />)
  const item = wrapper.find(CheckBox)
  const onchange = item.props().onChange
  if (onchange) {
    onchange({ target: { checked: true } } as React.ChangeEvent<
      HTMLInputElement
    >)
  }
  expect(mountToJson(wrapper)).toMatchSnapshot()
})
