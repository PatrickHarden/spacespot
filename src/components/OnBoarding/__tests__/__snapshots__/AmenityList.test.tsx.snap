// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing  1`] = `
<Styled(Component)
  amenities={
    Object {
      "op1": Object {
        "checked": true,
        "desc": "op1",
      },
    }
  }
>
  <StyledComponent
    amenities={
      Object {
        "op1": Object {
          "checked": true,
          "desc": "op1",
        },
      }
    }
    forwardedComponent={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bwzfXH",
          "isStatic": false,
          "lastClassName": "kyWUYx",
          "rules": Array [
            "
  min-height: 100px;
  margin-left: 10px;
  label {
    color: #828286;
  }
",
          ],
        },
        "displayName": "Styled(Component)",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bwzfXH",
        "target": [Function],
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
  >
    <Component
      amenities={
        Object {
          "op1": Object {
            "checked": true,
            "desc": "op1",
          },
        }
      }
      className="sc-bwzfXH kyWUYx"
    >
      <div
        className="sc-bwzfXH kyWUYx"
      >
        <Styled(Component)
          checked={true}
          key="op1"
          onChange={[Function]}
        >
          <StyledComponent
            checked={true}
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bdVaJa",
                  "isStatic": false,
                  "lastClassName": "fxjLYd",
                  "rules": Array [
                    "
  display: block;
  position: relative;
  cursor: pointer;
  padding-left: 25px;
  margin: 10px 0;
  input {
    position: absolute;
    opacity 0;
    cursor: pointer;
    width: 0;
    height: 0
  }
  span {
    position: absolute;
    top: 0;
    left: 0;
    height: 15px;
    width: 15px;
    border: 2px solid #B3DBE9;
    margin-top: 1px;
  }
  input:checked ~ span {
    background-color: transparent;
  }

  span:after {
    content: \\" \\";
    position: absolute;
    display: none;
  }

  input:checked ~ span:after {
    display: block;
  }

  span:after {
    left: 5px;
    top: 2px;
    width: 5px;
    height: 8px;
    border: solid black;
    border-width: 0 1px 2px 0;
    transform: rotate(45deg);
  }
",
                  ],
                },
                "displayName": "Styled(Component)",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-bdVaJa",
                "target": [Function],
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
            onChange={[Function]}
          >
            <Component
              checked={true}
              className="sc-bdVaJa fxjLYd"
              onChange={[Function]}
            >
              <label
                className="sc-bdVaJa fxjLYd"
              >
                op1
                <input
                  checked={true}
                  onChange={[Function]}
                  type="checkbox"
                />
                <span />
              </label>
            </Component>
          </StyledComponent>
        </Styled(Component)>
      </div>
    </Component>
  </StyledComponent>
</Styled(Component)>
`;
